test/tests/functional/pbs_power_provisioning_cray.py

```diff
@@ -449,15 +500,536 @@ def test_last_used_time(self):
+        for n in self.server.status(NODE):
+            if 'resources_available.PBScraynid' in n:
+                self.server.manager(MGR_CMD_SET, NODE,
+                                    {"poweroff_eligible": True}, n['id'])
+        a = {"power_on_off_enable": True,
+             "max_concurrent_nodes": "30", 'node_idle_limit': '30'}
+        self.modify_hook_config(attrs=a, id='PBS_power')
+        a = {'freq': 30}
+        self.server.manager(MGR_CMD_SET, PBS_HOOK, a, id='PBS_power',
+                            sudo=True)
+        a = {'enabled': 'True'}
+        self.server.manager(MGR_CMD_SET, PBS_HOOK, a, id='PBS_power',
+                            sudo=True)
+        t = int(time.time())
+        self.logger.info("Waiting for 4 mins to poweroff all the nodes")
+        time.sleep(240)
+        self.server.log_match(
+            "/opt/cray/capmc/default/bin/capmc node_off", starttime=t)
+        for n in self.server.status(NODE):
+            if 'resources_available.PBScraynid' in n:
+                self.server.expect(NODE, {'state': 'sleep'}, id=n['id'])
+        a = {"node_idle_limit": "1800", 'min_node_down_delay': '30'}
+        self.modify_hook_config(attrs=a, id='PBS_power')
+
+        jid1 = self.submit_job(1000, {'Resource_List.vnode': self.names[0],
+                                      'Resource_List.vntype': 'cray_compute'})
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+        t = int(time.time())
+        self.logger.info("Waiting for 10 mins to poweron the calandered nodes")
+        time.sleep(600)
+        self.server.log_match(
+            "/opt/cray/capmc/default/bin/capmc node_on", starttime=t)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        self.server.expect(NODE, {'state': 'job-exclusive'}, id=self.names[0])
+        a = {'queue_type': 'execution',
+             'started': 'True',
+             'enabled': 'True',
+             'priority': '150'}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='wq1')
+        t = int(time.time())
+        jid2 = self.submit_job(1000, {ATTR_queue: 'wq1',
+                                      'Resource_List.vntype': 'cray_compute'})
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid2)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid1)
+        self.scheduler.log_match("Job preempted by requeuing", starttime=t)
+        self.cleanup_power_on()
+
+    def test_last_used_time_node_sort_key(self):
+        """
+        Test last_used_time as node sort key.
+        """
+        self.server.manager(MGR_CMD_SET, SERVER, {
+                            'job_history_enable': 'True'})
+        i = 0
+        for n in self.server.status(NODE):
+            if 'resources_available.PBScraynid' in n:
+                if i > 1:
+                    self.server.manager(MGR_CMD_SET, NODE, {
+                        'state': 'offline'}, id=n['id'])
+                i += 1
```

Comment:
No need to keep adding to i.  Just set a boolean to True after the first pass through the list and check for it.
offline_vnode = False
...
if offline_vnode is True
...
offline_vnode = True

Reply:
I need 2 nodes and making rest all nodes as offline

Reply:
Oh, good point, this gives you two nodes, not just one.